ibis/backends/datafusion/__init__.py

```diff
@@ -114,23 +114,11 @@ def _get_schema_using_query(self, query: str) -> sch.Schema:
             pass
 
         try:
-            result = (
-                self.raw_sql(f"DESCRIBE {table.sql(self.name)}")
-                .to_arrow_table()
-                .to_pydict()
-            )
+            df = self.con.table(name)
         finally:
             self.drop_view(name)
-        return sch.Schema(
-            {
-                name: self.compiler.type_mapper.from_string(
-                    type_string, nullable=is_nullable == "YES"
-                )
-                for name, type_string, is_nullable in zip(
-                    result["column_name"], result["data_type"], result["is_nullable"]
-                )
-            }
-        )
+
+        return PyArrowSchema.to_ibis(df.schema())
```

Comment:
This turns out to be a **much** better strategy than trying to parse the string representations of arrow types (what `DESCRIBE t` produces) as SQL types.